### DISCUSSION TITLE
Is it possible to references ojs variables in html code blocks?

### DISCUSSION THREAD

**Discussion by jmhimara** [quarto-dev/quarto-cli#5719] [Q&A] [answered]
### Description

I know you can easily reference ojs variables in markdown, for example: 

````qmd

```{ojs}
viewof x = Inputs.button("Button")
```

${x}
````

However this doesn't seem to work within html cells. Is something like this possible, without resorting to DOM manipulation? 

For example (using `x` from above)?

````qmd

```{ojs}
viewof x = Inputs.button("Button")
```

```{=html}
<div>
<h2> Example HTML code block <h2>

Number of clicks: ${x}

<div>
```
````

**Comment by mcanouil** [comment]
You are talking about "html cells", but there is no such thing in your example.

Could you share a small "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by jmhimara** [reply]
Sure, I've edited my post to include an html code block.

**Reply by mcanouil** [reply]
Please do provide a complete Quarto document.
It's important to also include frontmatter you used to produce the document/error.
Note that this is explain in the discussion header when you opened it and in my message.
Thank you.

**Reply by jmhimara** [reply]
This is it, plus the frontmatter. There is no error. It's just the variables don't render in html in the same way they render in markdown. For instance, in the example below, the variables just appears as `${x}`.

````qmd
---
title: "Reproducible Quarto Document"
format: html
theme: cosmo
execute:
      echo: false
---

```{ojs}
viewof x = Inputs.button("Button")
```

```{=html}
<div>
<h2> Example HTML code block </h2>

Number of clicks: ${x}

<div>
```
````

**Reply by mcanouil** [reply]
You are telling Quarto/Pandoc to consider the `${x}` as raw code/text, especially as html, why would you expect it to not be read as is?

**Reply by jmhimara** [reply]
Sure, but is the alternative not possible? I.e. I.e. consider x as a variable?

**Reply by mcanouil** [reply]
Don't put it in raw block.
So reformat your block to make it appear outside the raw block.

**Reply by jmhimara** [reply]
I don't think that's possible for my case. I'm trying to build a rather complicated layout for a page that I don't think it can be done in markdown. I need to use raw html for this one.

**Reply by mcanouil** [reply, marked as answer]
It's possible using raw code blocks and raw inline code.
`` `<p>raw inline html</p>`{=html} followed by normal markup``

**Reply by jmhimara** [reply]
This could work. I think I should be able to split opening and closing html tags in different code blocks. 

Thanks.

**Comment by Xavi007** [comment]
If by "ojs variables" you are referring to variables used in Open Journal Systems (OJS), a scholarly publishing platform, and you want to reference them in HTML code blocks, it depends on the specific context and capabilities of OJS.

HTML itself does not have built-in support for variables or dynamic content. However, if OJS provides a templating system or scripting language that allows you to generate HTML code dynamically, you may be able to reference OJS variables within that system.

**Reply by mcanouil** [reply]
It's about Observable.js as stated in the discussion label 😉
